components/fxa-client/src/lib.rs

```diff
@@ -243,10 +254,28 @@ impl FirefoxAccount {
         }
     }
 
-    fn get_refresh_token(&self) -> Result<&str> {
+    fn get_refresh_token(&mut self) -> Result<&str> {
         match self.state.refresh_token {
             Some(ref token_info) => Ok(&token_info.token),
-            None => Err(ErrorKind::NoRefreshToken.into()),
+            None => {
+                log::info!("No refresh_token, attempting to recover via migration state");
+                // if we have no refresh token, check if this is the case of a failed migration
+                if self.state.in_flight_migration.is_some() {
+                    // if there's a pending migration state, then try to provision a refresh_token
+                    // by exchanging the session token for the refresh token
+                    self.helper_migration_network_methods();
```

Comment:
Is this wrong to assume we always want to do an in-flight-migration? I think we should do the `if is_some() migrate` before the `match self.state.refresh_token {`

Reply:
yeap there's a `if self.state.in_flight_migration.is_some() {`

Reply:
Sorry I wasn't really clear, I meant we could execute that block before the rest of the function.